src/System.Security.Cryptography.Pkcs/tests/CertLoader.cs

```diff
@@ -0,0 +1,150 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Test.Cryptography
+{
+    internal abstract partial class CertLoader
+    {
+        /// <summary>
+        /// Returns a freshly allocated X509Certificate2 instance that has a public key only. 
+        /// 
+        /// This method never returns null.
+        /// </summary>
+        public X509Certificate2 GetCertificate()
+        {
+            return new X509Certificate2(CerData);
+        }
+
+        /// <summary>
+        /// Attempts to return a freshly allocated X509Certificate2 instance that has a public and private key. Only use this method if your test
+        /// needs the private key to work. Otherwise, use GetCertificate() which places far fewer conditions on you.
+        /// 
+        /// The test must check the return value for null. If it is null, exit the test without reporting a failure. A null means
+        /// the test host chose to disable loading private keys.
+        /// 
+        /// If this method does return a certificate, the test must Dispose() it manually and as soon it no longer needs the certificate. 
+        /// Due to the way PFX loading works on Windows, failure to dispose may leave an artifact on the test machine's disk. 
+        /// </summary>
+        public X509Certificate2 TryGetCertificateWithPrivateKey()
+        {
+            if (PfxData == null)
+                throw new Exception("Cannot call TryGetCertificateWithPrivateKey() on this CertLoader: No PfxData provided.");
+
+            if (!_alreadySearchedMyStore)
+            {
+                // Machine config check: Make sure that a matching certificate isn't stored in the MY store. Apis such as EnvelopedCms.Decrypt() look in the MY store for matching certs (with no opt-out)
+                // and having our test certificates there can cause false test errors or false test successes.
+                //
+                // This may be an odd place to do this check but it's better than expecting every individual test to remember to do this.
+                CheckIfCertWasLeftInCertStore();
```

Comment:
This seems like the LoadFromStore mode would always fail, since it'll either be present (and throw here) or not (and fail in LoadFromStore).  Unless you're hiding them in obscure stores?